File: frontend/src/models/AdminBookRequest.ts
export interface AdminBookRequest {
  id?: number;
  title: string;
  author: string;
  description: string;
  copies: number;
  category: string;
  dataSource?: string;
  hasPdf?: boolean;
  hasImage?: boolean;
  imageUrl?: string;
  pdfFilename?: string;
  imageFilename?: string;
}

export interface AdminBookEditInfoResponse {
  hasPdf: boolean;
  hasImage: boolean;
  imageUrl?: string;
  pdfFilename?: string;
  imageFilename?: string;
}

File: spring-boot-library/src/main/java/com/luv2code/spring_boot_library/requestmodel/AdminBookRequest.java
package com.luv2code.spring_boot_library.requestmodel;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AdminBookRequest {
    private String title;
    private String author;
    private String description;
    private Integer copies;
    private String category;
}
